src/components/Analyze/AnalyzeList.tsx

```diff
@@ -15,32 +34,30 @@ export const AnalyzeList = () => {
 				</div>
 				<div className='overscroll-y-auto overflow-hidden w-full h-full'>
 					<ul className='flex flex-col flex-initial w-full h-full mt-5 gap-10 overflow-y-scroll pb-10'>
-						{consumptionCombinedData?.map((data) => {
-							return (
-								<li
-									key={data.id}
-									className='flex flex-col flex-initial text-black'
-								>
-									<div className='flex flex-1 justify-between'>
-										<div className='text-[#675f5f] text-[26px] font-medium'>
-											{data.category}
-										</div>
-										<div className='text-black text-2xl font-medium '>
-											{addComma(data.price)}
-										</div>
+						{consumptionCombinedData?.map((data) => (
+							<li
+								key={data.id}
+								className='flex flex-col flex-initial text-black'
+							>
+								<div className='flex flex-1 justify-between'>
+									<div className='text-[#675f5f] text-[26px] font-medium'>
+										{data.category}
 									</div>
-									<div className='flex-1 w-full h-[35px] bg-[#e8e8e8] rounded-[10px] shadow-analyze-chart'>
-										<div
-											className='h-[35px] rounded-[10px] shadow-analyze-progress-bar'
-											style={{
-												width: `${progressBarWidth ? progressBarWidth[data.id] : 0}%`,
-												backgroundColor: color[data.id % 4],
-											}}
-										/>
+									<div className='text-black text-2xl font-medium '>
+										{addComma(data.price)}
 									</div>
-								</li>
-							)
-						})}
+								</div>
+								<div className='flex-1 w-full h-[35px] bg-[#e8e8e8] rounded-[10px] shadow-analyze-chart'>
+									<div
+										className='h-[35px] rounded-[10px] shadow-analyze-progress-bar transition-[width] duration-400 ease-in-out'
+										style={{
+											width: animatedWidth[data.id] || '0%',
+											backgroundColor: chartColors[data.id % 4],
```

Comment:
나중에 색이 늘어난다면, 아래의 코드로 반복이 가능하겠네요.
```suggestion
backgroundColor: chartColors[data.id % chartColors.length],
```